### PR TITLE
Basketball Add Moved as an Optional Output of Activity

### PR DESCRIPTION
**Pull request by JohnDiebold**
This adds the moved activity in basketball (player moved to a different fantasy roster slot). It is an optional argument with default value of False so that existing usage is not affected.

### CODE REVIEW

**Comment by JohnDiebold** [comment]
@cwendt94 I've tested this with a couple of my leagues. If there's other testing that needs to be done or things needing changing let me know.

Just to add some context for why I would find this useful. I have a MySQL database for each fantasy teams roster on each day of the season. I use that to sum each teams projected points for the week (this is a points league). I'd like to store the time of the last activity and compare it to when the table was last updated to know if I need to update the table because someone moved a player from or to the bench. Right now the activities only returns add/drops, not moves.

File: espn_api/basketball/activity.py [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
             action = 'UNKNOWN'
             player = ''
+            position = ''
             msg_id = msg['messageTypeId']
             if msg_id == 244:
                 team = get_team_data(msg['from'])
             elif msg_id == 239:
                 team = get_team_data(msg['for'])
+            elif msg_id == 188:
+                if include_moved:
```
Comment (cwendt94) [inline espn_api/basketball/activity.py:18]: these checks can be moved to the top elif `msg_id == 188` and `and` together

File: espn_api/basketball/activity.py [hunk 2]
```diff
@@ -1,28 +1,43 @@
...
+                    if msg['to'] in POSITION_MAP:
+                        position = POSITION_MAP[msg['to']]
             else:
                 team = get_team_data(msg['to'])
             if msg_id in ACTIVITY_MAP:
-                action = ACTIVITY_MAP[msg_id]
+                if include_moved:
+                    action = ACTIVITY_MAP[msg_id]
+                else:
```
Comment (cwendt94) [inline espn_api/basketball/activity.py:26]: this can be a `elif msg_id != 188`

File: espn_api/basketball/league.py [hunk 3]
```diff
@@ -100,7 +100,7 @@ def recent_activity(self, size: int = 25, msg_type: str = None, offset: int = 0)
         headers = {'x-fantasy-filter': json.dumps(filters)}
         data = self.espn_request.league_get(extend='/communication/', params=params, headers=headers)
         data = data['topics']
-        activity = [Activity(topic, self.player_map, self.get_team_data) for topic in data]
+        activity = [act for act in (Activity(topic, self.player_map, self.get_team_data, include_moved=include_moved) for topic in data)if act and str(act).strip()]
```
Comment (cwendt94) [inline espn_api/basketball/league.py:103]: Why did this change other than adding `include_moved=include_moved`?

**Review by cwendt94 — COMMENTED** [review]
Just a couple comments and small suggestions

File: espn_api/basketball/activity.py (hunk 1 above)
Comment (JohnDiebold) [reply espn_api/basketball/activity.py]: Makes sense

File: espn_api/basketball/activity.py (hunk 2 above)
Comment (JohnDiebold) [reply espn_api/basketball/activity.py]: Also makes sense

File: espn_api/basketball/league.py (hunk 3 above)
Comment (JohnDiebold) [reply espn_api/basketball/league.py]: I don't remember what I was trying to do here, changed back to the original and added the include_moved=include_moved

**Comment by JohnDiebold** [comment]
Changes from all suggestions made. Thanks for the review!

**Pull request merged**